install/kubernetes/cilium/README.md

```diff
@@ -56,10 +56,11 @@ contributors across the globe, there is almost always someone available to help.
 | affinity | object | `{"podAntiAffinity":{"requiredDuringSchedulingIgnoredDuringExecution":[{"labelSelector":{"matchLabels":{"k8s-app":"cilium"}},"topologyKey":"kubernetes.io/hostname"}]}}` | Affinity for cilium-agent. |
 | agent | bool | `true` | Install the cilium agent resources. |
 | agentNotReadyTaintKey | string | `"node.cilium.io/agent-not-ready"` | Configure the key of the taint indicating that Cilium is not ready on the node. When set to a value starting with `ignore-taint.cluster-autoscaler.kubernetes.io/`, the Cluster Autoscaler will ignore the taint on its decisions, allowing the cluster to scale up. |
+| aksbyocni.enabled | bool | `false` | Enable AKS BYOCNI integration. Note that this is incompatible with AKS clusters not created in BYOCNI mode: use Azure integration (`azure.enabled`) instead. |
 | alibabacloud.enabled | bool | `false` | Enable AlibabaCloud ENI integration |
 | annotateK8sNode | bool | `false` | Annotate k8s node upon initialization with Cilium's metadata. |
 | autoDirectNodeRoutes | bool | `false` | Enable installation of PodCIDR routes between worker nodes if worker nodes share a common L2 network segment. |
-| azure.enabled | bool | `false` | Enable Azure integration |
+| azure.enabled | bool | `false` | Enable Azure integration. Note that this is incompatible with AKS clusters created in BYOCNI mode: use AKS BYOCNI integration (`aksbyocni.enabled`) instead. |
```

Comment:
Can we make `aksbyocni` under `azure` like `azure.byocni`? I don't see a need to have them as separate top-level values.

Reply:
Another option which I thought about was NOT to have a `aksbyocni` group at all, since all it does is set IPAM to Cluster Pool and disable tunneling. So we don't need a new mode at all and could simply have both options passed in correctly to the Helm chart in the docs, and the CLI handling it transparently for the CLI install (as it already does).

But we decided with @aanm that it was nice to advertise BYOCNI as an explicit alternative to Azure IPAM to the Helm values.

Reply:
:ok_hand: Makes sense, thanks for the explanation.